api/init/security/oauth/google.py

```diff
@@ -1,67 +1,69 @@
-    }
-}
 
-# Create an OAuth flow (Authorization Code) from client id and client secret
-flow = Flow.from_client_config(CLIENT_CONFIG, SCOPES)
-# Redirect the user back after login
-flow.redirect_uri = google_redirect_url
+def get_user_info(google_session: object):
+    """Gets user profile using OAuth session."""
 
+    user_profile = google_session.get(USER_PROFILE_URI).content.decode('utf-8')
+    user_profile = json.loads(user_profile)
+    print(user_profile)
 
-def get_user_info(oauth_token: str):
-    """Gets the user profile of the user after login using the corresponding OAuth token"""
-    request_url = USER_PROFILE.format(oauth_token)
-    response = requests.get(request_url)
-    return json.loads(response.text)
+    return user_profile
 
 
 def register_google_oauth(namespace: Namespace):
-    """Registers all endpoints used for Google OAuth authentication"""
+    """Registers all endpoints used for Google OAuth authentication."""
 
     @namespace.route('/security/oauth/google')
     @namespace.doc()
     class GoogleOAuth(Resource):
-        """Defines the resource to redirect the user to the Google OAuth page"""
+        """Defines resource to redirect user to Google OAuth page."""
 
         def get(self):
-            """Redirects the user to the Google OAuth page"""
-            url, _ = flow.authorization_url()
+            """Redirects user to Google OAuth page."""
+
+            google_session = OAuth2Session(client_id, redirect_uri=redirect_uri, scope=SCOPE)
+            url, state = google_session.authorization_url(AUTHORIZATION_URI, access_type="offline", prompt="select_account")
+
+            # State is used to prevent CSRF, keep this for later.
+            session['oauth_state'] = state
+            print(url)
             return redirect(url)
 
     @namespace.route('/security/oauth/google/callback')
     @namespace.doc()
     class GoogleOAuthCallback(Resource):
-        """Defines the resource to handle the callback from Google OAuth"""
+        """Defines resource to handle callback from Google OAuth."""
 
         def get(self):
-            """Handles the Google OAuth callback and returns the token in the cookie"""
-            code = request.args.get('code')
-            token = flow.fetch_token(code=code)
-            user_info = get_user_info(token['access_token'])
-            jwt = get_jwt_token(
-                TokenType.Google, user_info['email'], user_info, token)
-            return get_token_redirect_response(jwt)
+            """Handles Google OAuth callback and fetch user access token."""
+
+            # google_session = OAuth2Session(client_id, state=session['oauth_state'])
+            google_session = OAuth2Session(client_id, redirect_uri=redirect_uri, scope=SCOPE)
+            print(request.url)
+            token = google_session.fetch_token(TOKEN_URI, client_secret=client_secret, authorization_response=request.url)
+
+            # Persist token in session
```

Comment:
Same here as in Github OAuth. I think we're not gonna need the token. Also I would remove the print and the comment about the google_session.

Reply:
Agreed, done.